docs/enhancements/aws-private-link.md

```diff
@@ -0,0 +1,553 @@
+
+### Manual walk-through
+
+Here is a manual walk through of all the steps that need to be taken today
+to setup the required environment.
+
+So let's assume we have a public Route53 HostedZone `hub01.osdprivate.io`. And
+the computes DNS name for the endpoint service is `api.hash123.hub01osdprivate.io`.
+
+Create the manifests for install.
+```shell=bash
+$ openshift-install --dir install_dir01 create manifests
+```
+
+Add the APIServer and Secret object for setting up apiserver SNI.
+```shell=bash
+$ cat > install_dir01/manifests/apiserver.yaml << EOF
+apiVersion: config.openshift.io/v1
+kind: APIServer
+metadata:
+  name: cluster
+spec:
+  servingCerts:
+    namedCertificates:
+    - names:
+      - api.hash123.hub01osdprivate.io
+      servingCertificate:
+       name: apiserver-private-link-ingress
+EOF
+$ cat > install_dir01/manifests/secret-apiserver-private-link-ingress.yaml << EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: openshift-config
+  name: apiserver-private-link-ingress
+tls.key: <the serving certificate key here>
+tls.crt: <the serving certificate here>
+EOF
+```
+
+Create infrastructure resources only.
+```shell=bash
+$ openshift-install --dir install_dir01 create cluster --no-wait-only-infra
+```
+
+At this stage the installer has created the NLB for the apiserver and the
+installation is already in progress in the bootstrap ec2instance. We can use
+the installer tags to figure out the apiserver's internal NLB for the cluster.
+
+Create an Endpoint service in AWS account of the cluster.
+```shell=bash
+$ AWS_PROFILE=customer aws ec2 create-vpc-endpoint-service-configuration \
+--no-acceptance-required \
+--private-dns-name api.hash123.hub01osdprivate.io \
+--network-load-balancer-arns <nlb arn here>
+```
+```shell=bash
+$ aws ec2 modify-vpc-endpoint-service-permissions \
+--add-allowed-principals '["arn:aws:iam::123456789012:root"]' 
+```
+
+Extract the DNS verification information from the create.
+```shell=bash
+$ DOMAIN_VERIFICATION_NAME=$(jq '.ServiceConfigurations[0].PrivateDnsNameConfiguration.Name')
+$ DOMAIN_VERIFICATION_VALUE=$(jq '.ServiceConfigurations[0].PrivateDnsNameConfiguration.Value')
+```
+
+Let's create the TXT records in HostedZone to complete the verification.
+NOTE: This is happening in the hub AWS account.
+```json
+{
+  "HostedZoneId": "$HOSTED_ZONE_ID",
+  "ChangeBatch": {
+    "Comment": "Add TXT record for domain verification",
+    "Changes": [
+      {
+        "Action": "UPSERT",
+        "ResourceRecordSet": {
+          "Name": "$DOMAIN_VERIFICATION_NAME.api.hash123.hub01osdprivate.io",
+          "Type": "TXT",
+          "RecordSets": [{
+            "Value": "$DOMAIN_VERIFICATION_NAME"
+          }]
+        }
+      }
+    ]
+  }
+}
+```
+
+Now trigger the verifiation of the domain name manually.
+```shell=bash
+$ AWS_PROFILE=customer aws ec2 start-vpc-endpoint-service-private-dns-verification
+```
+
+At this point the Endpoint service is ready for use. Next create an Endpoint
```

Comment:
VPC Endpoint Service and VPC Endpoints can of Interface type or Gateway type. Since we are using private NLB as ingress, Interface type is being used.

Updated to be a little more consistent and clear.